intelmq/bots/parsers/generic/parser_csv.py

```diff
@@ -98,9 +98,14 @@ def parse(self, report):
         # ignore lines having mix of spaces and tabs only
         raw_report = re.sub(r'(?m)^[ \t]*\n?', '', raw_report)
         # skip header
-        if self.skip_header:
-            self.tempdata.append(raw_report[:raw_report.find('\n')])
-            raw_report = raw_report[raw_report.find('\n') + 1:]
+        if bool(self.skip_header):
+            split = raw_report.splitlines()
+            if isinstance(self.skip_header, int):
```

Comment:
Instead of checking the type of the parameter, converting it to int is much easier.